packaging_automation/update_package_properties.py

```diff
@@ -2,36 +2,62 @@
 import re
 from dataclasses import dataclass
 from datetime import datetime
+from enum import Enum
+import os
 
 import pathlib2
 import string_utils
 from github import Github
-from parameters_validation import (no_whitespaces, non_blank, non_empty, non_negative, validate_parameters,
+from parameters_validation import (non_blank, non_empty, validate_parameters,
                                    parameter_validation)
 
 from .common_tool_methods import (find_nth_matching_line_and_line_number, find_nth_occurrence_position,
-                                  get_project_version_from_tag_name, get_template_environment)
-
+                                  get_project_version_from_tag_name, get_template_environment, initialize_env, run,
+                                  create_pr, remove_cloned_code)
 from .common_validations import (is_version, is_tag)
+import uuid
 
 BASE_PATH = pathlib2.Path(__file__).parent.absolute()
-
-project_name_suffix_dict = {"citus": "citus", "citus-enterprise": "citus",
-                            "pg-auto-failover": "", "pg-auto-failover-enterprise": "", "pg-cron": "", "pg-xn": ""}
+REPO_OWNER = "citusdata"
+PROJECT_NAME = "packaging"
+
+
+@dataclass()
+class ProjectDetails:
+    name: str
+    version_suffix: str
+    github_repo_name: str
+    changelog_project_name: str
+    packaging_branch: str
+
+
+class SupportedProjects(Enum):
+    citus = ProjectDetails(name="citus", version_suffix="citus", github_repo_name="citus",
+                           changelog_project_name="citus", packaging_branch="all-citus")
+    citus_enterprise = ProjectDetails(name="citus-enterprise", version_suffix="citus",
+                                      github_repo_name="citus-enterprise", changelog_project_name="citus-enterprise",
+                                      packaging_branch="all-enterprise")
+    pg_auto_failover = ProjectDetails(name="pg-auto-failover", version_suffix="",
+                                      github_repo_name="pg_auto_failover", changelog_project_name="pg_auto_failover",
+                                      packaging_branch="all-pgautofailover")
+    pg_auto_failover_enterprise = ProjectDetails(name="pg-auto-failover-enterprise", version_suffix="",
+                                                 github_repo_name="citus-ha",
+                                                 changelog_project_name="pg_auto_failover-enterprise",
+                                                 packaging_branch="all-pgautofailover-enterprise")
 
 
 @parameter_validation
 def is_project_changelog_header(header: str):
     if not header:
         raise ValueError("header should be non-empty and should not be None")
-    if not re.match(r"^### \w+\sv\d+\.\d+\.\d+\s\(\w+\s\d+,\s\d+\)\s###$", header):
+    if not re.match(r"^### \w+[-]?\w+\sv\d+\.\d+\.\d+\s\(\w+\s\d+,\s\d+\)\s###$", header):
```

Comment:
can we have comment to make it clear what does that regex catches, like:
`# an example matching string is "citus-X Y foo-bar"`

Reply:
Fixed